lib/plugins/aws/package/compile/events/stream/index.js

```diff
@@ -12,6 +12,35 @@ class AwsCompileStreamEvents {
     };
   }
 
+  isValidStackImport(variable) {
+    if (Object.keys(variable).length !== 1) {
+      return false;
+    }
+    if (
+      variable['Fn::ImportValue'] &&
+      (variable['Fn::ImportValue']['Fn::GetAtt'] || variable['Fn::ImportValue'].Ref)
+    ) {
+      return false;
+    }
+    const intrinsicFunctions = ['Fn::ImportValue', 'Ref', 'Fn::GetAtt', 'Fn::Sub', 'Fn::Join'];
+    return intrinsicFunctions.some(func => _.has(variable, func));
+  }
+
+  resolveInvalidDestinationPropertyErrorMessage(functionName, property) {
+    return [
+      `Missing or invalid ${property} property for on failure destination`,
+      ` in function "${functionName}"`,
+      ' The correct syntax is: ',
+      ' destinationConfig: ',
+      '   onFailure: ',
+      '     destination: ',
+      '       arn: resource-arn',
+      '       type: (sns/sqs)',
+      ' OR an object with arn and type',
+      ' Please check the docs for more info.',
+    ].join('');
```

Comment:
If I read correctly, all this message will be concatenated into one line, which will make instruction not readable.

It'll probably be good to ensure that at least configuration instruction is multiline.